src/Symfony/Bridge/Doctrine/Security/User/EntityUserProvider.php

```diff
@@ -110,6 +108,36 @@ public function refreshUser(UserInterface $user)
      */
     public function supportsClass($class)
     {
-        return $class === $this->class || is_subclass_of($class, $this->class);
+        return $class === $this->getClass() || is_subclass_of($class, $this->getClass());
+    }
+
+    private function getEntityManager()
+    {
+        return $this->registry->getManager($this->managerName);
+    }
+
+    private function getRepository()
+    {
+        return $this->getEntityManager()->getRepository($this->getClass());
+    }
+
+    private function getClass()
+    {
+        if (null === $this->class) {
+            $class = $this->classOrAlias;
+
+            if (false !== strpos($class, ':')) {
+                $class = $this->getClassMetadata()->getName();
+            }
+
+            $this->class = $class;
+        }
+
+        return $this->class;
+    }
+
+    private function getClassMetadata()
+    {
+        return $this->getEntityManager()->getClassMetadata($this->getClass());
```

Comment:
Wow, good eye. In fact, I should only use `getClass()` where we actually need a valid PHP class name. I've updated to do that (otherwise, just use the class/alias that was passed in).